### PR TITLE
Update the orderBy filtering mechanism for OR queries.

### PR DESCRIPTION
**Pull request by ehsannas**

### CODE REVIEW

File: firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java [hunk 1]
```diff
@@ -392,7 +392,12 @@ private boolean matchesFilters(Document doc) {
 
   /** A document must have a value for every ordering clause in order to show up in the results. */
   private boolean matchesOrderBy(Document doc) {
-    for (OrderBy order : explicitSortOrder) {
+    // We must use `getOrderBy` here, not `explicitSortOrder`. For queries without disjunctions,
```
Comment (wu-hui) [inline firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java:395]: I am confused by how this change makes the fix. Can you give a better explanation or some example in this comment?

Comment (ehsannas) [reply firebase-firestore/src/main/java/com/google/firebase/firestore/core/Query.java]: I have attempted to improve this comment with an example. Please let me know if it's still unclear. Thanks

**Review by wu-hui — APPROVED** [review]
Yes, much better with an example!

**Pull request merged**